main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -288,6 +289,21 @@ object Unification {
       unifyTypeM(ts, loc)
   }
 
+  /**
+    * Pairwise unifies the types in `tpes1` and `tpes2` with the corresponding source locations `locs`.
+    *
+    * Assumes that `tpes1`, `tpes2`, and `locs` have the same length.
+    */
+  def pairwiseUnifyM(tpes1: List[Type], tpes2: List[Type], locs: List[SourceLocation])(implicit flix: Flix): InferMonad[Unit] =
```

Comment:
We should communicate in the signature or function title that this is an `expectTypeM` rather than just a `unifyTypeM`